src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java

```diff
@@ -143,15 +146,25 @@ public Code retrieveCode(String code, String codeSystemName, String version, Str
     if (codeSystem == null) {
       return null;
     }
-    String codeJson = fhirTerminologyServiceWebClient.getCodeResource(code, codeSystem, apiKey);
+    String codeJson = fhirTerminologyServiceWebClient.getCodeResource(codeName, codeSystem, apiKey);
     Parameters parameters = fhirContext.newJsonParser().parseResource(Parameters.class, codeJson);
-    return Code.builder()
-        .name(code)
-        .codeSystem(codeSystemName)
-        .version(version)
-        .display(parameters.getParameter("display").getValue().toString())
-        .codeSystemOid(parameters.getParameter("Oid").getValue().toString())
-        .build();
+    Code code =
+        Code.builder()
+            .name(codeName)
+            .codeSystem(codeSystemName)
+            .version(version)
+            .display(parameters.getParameter("display").getValue().toString())
+            .codeSystemOid(parameters.getParameter("Oid").getValue().toString())
+            .build();
+    String codePath =
+        TerminologyServiceUtil.buildCodePath(
+            code.getCodeSystem(), code.getVersion(), code.getName());
+    // TODO: this is ridiculous
```

Comment:
Can you clarify why this is ridiculous and maybe suggest a future solution?

Reply:
we are using new FHIR API to get the code details but because the new API doesn't return just that one field(status) we have to go back to old API to get just that one field. Because of this limitation, we are polluting our code plus double API call.  My solution is to push them to return that one field in new API.